module/zfs/qat_crypt.c

```diff
@@ -548,6 +587,8 @@ qat_checksum(uint64_t cksum, uint8_t *buf, uint64_t size, zio_cksum_t *zcp)
 	for (i = 0; i < page_num; i++)
 		kunmap(in_pages[i]);
 
+	/* wait session lock release before remove it */
+	symsessionwait(cy_session_ctx);
```

Comment:
I'd be interested to know what the performance impact is here.

Reply:
@cfzhu  Suggest not do it for now in this PR, will see if we can handle it in the QAT driver with a better way.